apps/yapms/src/routes/app/[country]/[map]/[year]/initialize/ApplyPanZoom.ts

```diff
@@ -3,13 +3,33 @@ import panzoom, { type PanZoom } from 'panzoom';
 function applyPanZoom(mapBind: HTMLDivElement): PanZoom | undefined {
 	const svg = mapBind.querySelector<SVGElement>('svg'); //We can do this since the only child of node is the map svg itself.
 	if (svg) {
-		return panzoom(svg, {
+		const panzoomInstance = panzoom(svg, {
 			minZoom: 0.5,
 			maxZoom: 100,
 			smoothScroll: false,
 			autocenter: true,
 			zoomDoubleClickSpeed: 1
 		});
+		if (svg.classList.contains('autoBorders')) {
+			adjustStroke(svg, panzoomInstance.getTransform().scale);
+			panzoomInstance.on('zoom', (e: PanZoom) => {
+				adjustStroke(svg, e.getTransform().scale);
+			});
+		}
+		return panzoomInstance;
+	}
+}
+
+function adjustStroke(svg: SVGElement, scale: number) {
+	const initStroke = svg.hasAttribute('auto-border-stroke')
+		? Number(svg.getAttribute('auto-border-stroke'))
+		: 0.5;
+	const strokeUpper = Number(svg.getAttribute('auto-border-upper'));
+	const newStroke = initStroke / scale;
+	if (strokeUpper && newStroke > strokeUpper) {
```

Comment:
If strokeUpper is 0 it will allow for unlimited stroke width.

Reply:
This is fine-- just remove the attributes in that case for no stroke

Reply:
But if someone sets the upper limit to zero, the software should cap the limit at zero. From an end user's perspective, when they are trying to make a map, they won't understand why setting the upper limit to zero results in unlimited upper limit (without reading the source code).

Reply:
In the case that no upper limit is set the code will return NaN now (instead of the Number(undefined) == 0) and won't pass the check. 0 will now result in no stroke